File: src/sql/postgres/protocol/CopyOutResponse.zig
const CopyOutResponse = @This();

pub fn decodeInternal(this: *@This(), comptime Container: type, reader: NewReader(Container)) !void {
    _ = reader;
    _ = this;
    bun.Output.panic("TODO: not implemented {s}", .{bun.meta.typeBaseName(@typeName(@This()))});
}

pub const decode = DecoderWrap(CopyOutResponse, decodeInternal).decode;

const bun = @import("bun");
const DecoderWrap = @import("./DecoderWrap.rust").DecoderWrap;
const NewReader = @import("./NewReader.rust").NewReader;

File: src/sql/postgres/protocol/CopyOutResponse.rs
use super::new_reader::NewReader;
use crate::postgres::AnyPostgresError;

pub struct CopyOutResponse;

impl CopyOutResponse {
    // COPY TO is not supported yet.
    pub fn decode_internal<Container: super::new_reader::ReaderContext>(
        &mut self,
        reader: NewReader<Container>,
    ) -> Result<(), AnyPostgresError> {
        drop(reader);
        let _ = self;
        bun_core::output::panic(format_args!("TODO: not implemented {}", "CopyOutResponse",));
    }
}

impl CopyOutResponse {
    pub fn decode<Container: super::new_reader::ReaderContext>(
        &mut self,
        context: Container,
    ) -> Result<(), AnyPostgresError> {
        self.decode_internal(NewReader { wrapped: context })
    }
}
